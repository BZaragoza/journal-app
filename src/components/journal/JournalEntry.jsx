import React from 'react'
import moment from 'moment'
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';
import { trimAndFormatString } from '../../helpers/trimAndFormatString';

const JournalEntry = ({ id, date, title, body, url }) => {

  const dispatch = useDispatch();


  const noteDate = moment(date);

  const handleActiveEntry = () => {
    dispatch( activeNote(id, {date, title, body, url})  )
  }

  return (
    <div 
      className="journal__entry animate__animated animate__fadeIn animate__faster"
      onClick={ handleActiveEntry }
    >
      
      {
        url &&
          <div 
          className="journal__entry-picture"
          style={{
            backgroundSize: 'cover',
            backgroundImage: `url(${ url })`
          }}
        ></div>
      }

      <div className="journal__entry-body">
        <p className="journal__entry-title">{ title }</p>
        <p className="journal__entry-content">{ trimAndFormatString( body ) }</p>
      </div>

      <div className="journal_entry-datebox">
        <span>{ noteDate.format('dddd') }</span>
        <h4>{ noteDate.format('D') }</h4>
      </div>

    </div>
  )
}

export default JournalEntry
