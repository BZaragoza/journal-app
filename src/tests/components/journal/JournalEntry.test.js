import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import JournalEntry from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';



const mockStore = configureStore([thunk]);
const store = mockStore( {} );
store.dispatch = jest.fn();

const note = {
  id: 0,
  date: 0,
  title: 'Hola',
  body: 'Mundo',
  url: 'https://someplace.com/pic.jpg'
}


const wrapper = mount(
  <Provider store={ store }>
    <JournalEntry { ...note } />
  </Provider>
)


describe('Tests on <JournalEntry />', () => {
  
  test('should render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should trigger activeNote', () => {
    
    wrapper.find('.journal__entry').prop('onClick')();

    expect( store.dispatch ).toHaveBeenCalledWith( 
      activeNote(note.id, {...note} ) 
    );

  })
  
  

})
