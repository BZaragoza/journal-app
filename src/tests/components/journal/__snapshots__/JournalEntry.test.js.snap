// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <JournalEntry /> should render correctly 1`] = `
<div
  className="journal__entry animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://someplace.com/pic.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hola
    </p>
    <p
      className="journal__entry-content"
    >
      Mundo
    </p>
  </div>
  <div
    className="journal_entry-datebox"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31
    </h4>
  </div>
</div>
`;
